src/main/java/spireMapOverhaul/zones/smithsFolly/SmithsFolly.java

```diff
@@ -0,0 +1,179 @@
+        //Replace all shop rooms with monster rooms
+        for (MapRoomNode node : this.nodes) {
+            if(node.room != null && ShopRoom.class.equals(node.room.getClass())) {
+                node.setRoom(new MonsterRoom());
+            }
+        }
+    }
+
+    @Override
+    public boolean allowSideConnections() {
+        return false;
+    }
+
+    @Override
+    public boolean allowAdditionalPaths() {
+        return false;
+    }
+
+    @Override
+    public void onEnter() {
+        upgradeRarestCard();
+    }
+
+    @Override
+    public void onExit() {
+        upgradeRarestCard();
+    }
+
+    @Override
+    public void atPreBattle() {
+        if (!AbstractDungeon.getCurrMapNode().hasEmeraldKey) {
+            AbstractRoom abstractRoom = AbstractDungeon.getCurrRoom();
+
+            int modifierIndex = AbstractDungeon.miscRng.random(0, 3);
+            switch (modifierIndex) {
+                case 0: {
+                    for (final AbstractMonster m : abstractRoom.monsters.monsters) {
+                        AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(m, m, new StrengthPower(m, AbstractDungeon.actNum + 1), AbstractDungeon.actNum + 1));
+                    }
+                    break;
+                }
+                case 1: {
+                    for (final AbstractMonster m : abstractRoom.monsters.monsters) {
+                        AbstractDungeon.actionManager.addToBottom(new IncreaseMaxHpAction(m, 0.25f, true));
+                    }
+                    break;
+                }
+                case 2: {
+                    for (final AbstractMonster m : abstractRoom.monsters.monsters) {
+                        AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(m, m, new MetallicizePower(m, AbstractDungeon.actNum * 2 + 2), AbstractDungeon.actNum * 2 + 2));
+                    }
+                    break;
+                }
+                case 3: {
+                    for (final AbstractMonster m : abstractRoom.monsters.monsters) {
+                        AbstractDungeon.actionManager.addToBottom(new ApplyPowerAction(m, m, new RegenerateMonsterPower(m, 1 + AbstractDungeon.actNum * 2), 1 + AbstractDungeon.actNum * 2));
+                    }
+                    break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public RewardItem getAdditionalReward() {
+        if (!AbstractDungeon.getCurrMapNode().hasEmeraldKey && AbstractDungeon.getCurrRoom() instanceof MonsterRoomElite) {
+            if (!Settings.hasEmeraldKey) {
+                return new RewardItem(null, RewardItem.RewardType.EMERALD_KEY);
+            }
+        }
+        return null;
+    }
+
+    private void upgradeRarestCard() {
+        AbstractPlayer player = AbstractDungeon.player;
+        if (player == null) return;
+
+        // Define the rarity order
+        List<AbstractCard.CardRarity> rarityOrder = Arrays.asList(
+                AbstractCard.CardRarity.SPECIAL,
```

Comment:
I understand the logic but it'd be more intuitive if special were above basic. People expect rare cards to be the rarest ones